content/en/docs/concepts/services-networking/network-policies.md

```diff
@@ -312,6 +323,60 @@ The value of the label is the namespace name.
 While NetworkPolicy cannot target a namespace by its name with some object field, you can use the
 standardized label to target a specific namespace.
 
+## Pod lifecycle
+
+A NetworkPolicy is applied to a selected pod, when isolation rules are enforced
+for that pod. At that point pod can never establish an outbound connection or receive
+an inbound connection that is not allowed by the NetworkPolicy. Allow rules will be applied
+eventually after the isolation rules (that includes applying both at the same time).
+That means, that once the NetworkPolicy is applied to a pod, the pod should be able to handle situations,
+when it can't establish a connection that is allowed by an existing NetworkPolicy for some time.
+
+Every NetworkPolicy will be applied to all selected pods eventually, but there is no
+way to tell from the existing Kubernetes API when exactly the network policy is applied to a specific
+pod. The only way to check if the network policy is applied to a pod, is to run a connectivity check.
```

Comment:
I'd remove the word "existing" since the discussions around NetworkPolicyStatus led to a consensus that there is no good way to ever do this from the Kubernetes API, particularly given the "distributed" thing where different pods may have different perceptions of the current status.

As for whether "connectivity check" is reliable; it's reliable if and only if you do the connectivity check _from that pod_. So maybe we could suggest that if a pod can't be rewritten to be resilient to lack-of-connectivity, that you could add an `initContainer` to it that checks that it has the connectivity it needs, so that things will be working as expected when the main containers start.